paperqa/clients/semantic_scholar.py

```diff
@@ -225,9 +225,16 @@ async def s2_title_search(
             HTTPStatus.NOT_FOUND: DOINotFoundError(f"Could not find DOI for {title}.")
         },
     )
-
-    if authors and not s2_authors_match(authors, data["data"][0]):
-        raise DOINotFoundError(f"Could not find DOI for {title} - author disagreement.")
+    try:
+        if authors and not s2_authors_match(authors, data=data["data"][0]):
+            raise DOINotFoundError(
+                f"Could not find DOI for {title} - author disagreement."
+            )
+    except KeyError as exc:  # Very rare, but "data" may not be in data
+        raise DOINotFoundError(
```

Comment:
@mskarlin can you confirm you think this is the right action to take here?

Reply:
for sure! nice catch here